### ISSUE TITLE
WebDavFileProvider example Fetching Contents of File

### ISSUE THREAD

**Issue by CPiersigilli** [amosavian/FileProvider#33]
This is portion of code for os x:
```
@IBAction func readText(_ sender: NSButton) {
        let user = "xxx@xxxxx.xx"
        let password = "xxxxxxxx"
        
        let credential = URLCredential(user: user, password: password, persistence: .forSession)
        let myUrlWebdav = URL(string: "https://webdav.pcloud.com")
        let documentsProvider = WebDAVFileProvider(baseURL: myUrlWebdav!, credential: credential)
        let path = "Cantiere 11.txt"

        documentsProvider?.contents(path: path, completionHandler: {
            contents, error in
            if error == nil {
            if let contents = contents {
                print(String(data: contents, encoding: .utf8))
                }
            }
            else
            {
                print("Error: \(error)")
            }
        })
    }
```
The error is:

> (Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLKey=https://webdav.pcloud.com/Cantiere%2011.txt, NSLocalizedDescription=cancelled, NSErrorFailingURLStringKey=https://webdav.pcloud.com/Cantiere%2011.txt})

Why?

**Comment by amosavian** [comment]
Short answer: because you don't retain `documentsProvider`. 

When function reaches end, it will release all variables defined inside, including `documentsProvider`. While because of async nature of all providers it will happen before fetching data.

if you want to keep tasks around, you may set `fileProviderCancelTasksOnInvalidating` global variable to false.

**Comment by CPiersigilli** [comment]
I change  `fileProviderCancelTasksOnInvalidating` to false in your `FileProvider.swift` and now the code work fine, but how to change this variable and where to block the tasks.

**Comment by amosavian** [comment]
`fileProviderCancelTasksOnInvalidating` is a global variable and you can change it in your code anytime. You don't have to change anything is FileProvider's source code.

You should know how Automatic Reference Counting and memory management works in swift:
Short answer: define it as a global variable or as member(property) of your class
TL;DR: https://www.raywenderlich.com/134411/arc-memory-management-swift

**Comment by CPiersigilli** [comment]
I test my code with version (0.15.0) and work fine without change `fileProviderCancelTasksOnInvalidating` to false in your `FileProvider.swift`.
However, today, I will try to read the post "`ARC and Memory Management in Swift`", but I do not know if I will understand.

**Comment by amosavian** [comment]
If you think the problem is solved you can close this issue. You can reopen it anytime you have further question on this problem.

**Comment by CPiersigilli** [comment]
I do not know why, but doing other tests, the program no longer works, even changing the `fileProviderCancelTasksOnInvalidating` to `False` in your `FileProvider.swift`.
The `error` for `documentsProvider?.contents` is: `Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLKey=https://webdav.pcloud.com/Text/newFile1.txt, NSLocalizedDescription=cancelled, NSErrorFailingURLStringKey=https://webdav.pcloud.com/Text/newFile1.txt}`
The `error` for `webdavFile?.create(folder:` is `Method Not Allowed`, because the folder exits and the `error` for `webdavFile?.writeContents(path:` is `Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLKey=https://webdav.pcloud.com/Text/newFile1.txt, NSLocalizedDescription=cancelled, NSErrorFailingURLStringKey=https://webdav.pcloud.com/Text/newFile1.txt}`
I hope you manage to make it work well, because `FileProvider` is very easy to use and has everything that I would serve to my app.
Thanks to the passion and commitment you are making to prepare this library.
I will continue to study and perform tests, although I'm not a professional programmer.

**Comment by CPiersigilli** [comment]
For a probable XCode bug, `fileProviderCancelTasksOnInvalidatin` is always `TRUE` even if I set it to `FALSE`, that is why it does not work, but I do not know how.
Tomorrow I will try again.
